pulp_rpm/app/modulemd.py

```diff
@@ -211,3 +218,32 @@ def parse_modular(file):
             logging.warning(f"Unknown modular document type found: {parsed_data.get('document')}")
 
     return modulemd_all, modulemd_defaults_all, modulemd_obsoletes_all
+
+
+class ModularYamlLoader(yaml.SafeLoader):
+    """Custom Loader that preserve unquoted float in specific fields"""
+
+    # Field to preserve (will bypass yaml casting)
+    PRESERVED_FIELDS = ("name", "stream", "version", "context", "arch")
```

Comment:
I believe this ought to handle the `stream` field on ModulemdDefaults and ModulemdObsoletes as well, but we ought to make sure it's tested

https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/models/modulemd.py#L101

We'll talk about this a bit more on a call